src/middlewares/cors.rs

```diff
@@ -0,0 +1,51 @@
+use std::pin::Pin;
+
+use futures::{FutureExt, TryFutureExt, future};
+use gotham::handler::HandlerFuture;
+use gotham::hyper::StatusCode;
+use gotham::hyper::header::{HeaderMap, ORIGIN};
+use gotham::middleware::Middleware;
+use gotham::prelude::*;
+use gotham::state::State;
+
+use std::env;
+
+#[derive(Clone, NewMiddleware)]
+pub struct CorsMiddleware;
+
+impl Middleware for CorsMiddleware {
+    fn call<Chain>(self, state: State, chain: Chain) -> Pin<Box<HandlerFuture>>
+        where
+            Chain: FnOnce(State) -> Pin<Box<HandlerFuture>>,
+        {
+            let allowed_origins = env::var("WHITELIST").unwrap_or_else(|_| "*".to_string());
+
+            let header_map = HeaderMap::borrow_from(&state);
+
+            let requests_origin = match header_map.get(ORIGIN) {
+                Some(origin) => origin.to_str().unwrap().to_string(),
+                None => "".to_string(),
```

Comment:
Not sure if this it's better than panic. If the request doesn't have an ORIGIN we can not check properly. Furthermore, later on line 31 `""` can be matched incorrectly so incorrect origin can be allowed.

Reply:
Do you suggest avoiding requests without the correct origin header?

Reply:
My issue is more with using `""` as default